src/Raven.Server/Documents/BackgroundWorkHelper.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Linq;
+using Raven.Client.ServerWide;
+using Raven.Server.Documents.DataArchival;
+using Raven.Server.Documents.Expiration;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents;
+
+public static class BackgroundWorkHelper
+{
+    public static bool CheckIfNodeIsFirstInTopology(ExpirationStorage.ExpiredDocumentsOptions expiredDocumentsOptions)
+    {
+        return CheckIfFirstInTopologyInternal(expiredDocumentsOptions.DatabaseTopology, expiredDocumentsOptions.NodeTag);
+    }
+
+    public static bool CheckIfNodeIsFirstInTopology(DataArchivalStorage.ArchivedDocumentsOptions archivedDocumentsOptions)
+    {
+        return CheckIfFirstInTopologyInternal(archivedDocumentsOptions.Topology, archivedDocumentsOptions.NodeTag);
+    }
+
+    private static bool CheckIfFirstInTopologyInternal(DatabaseTopology topology, string nodeTag)
```

Comment:
Let's remove the above two methods accepting `ExpiredDocumentsOptions / ArchivedDocumentsOptions` as an argument and have a single method

`public static bool ShouldHandleWorkOnCurrentNode(DatabaseTopology topology, string nodeTag)`.

This way the helper remains more generic and not tied to specific implementation like the expiration of the data archival

Reply:
done